soda-consumer-scala/src/main/scala/com/socrata/soda2/consumer/impl/LegacyRowDecoder.scala

```diff
@@ -27,12 +26,15 @@ private[consumer] class LegacyRowDecoder(datasetBase: URI, rawSchema: Map[Column
 }
 
 private[consumer] object LegacyRowDecoder {
-  val conversionForType: ((ColumnName, String)) => (ColumnName, SodaType) = { case (col, rawType) =>
-    col -> typeMap.getOrElse(rawType, throw new UnknownTypeException(rawType))(col.toString)
+  val conversionForType: PartialFunction[(ColumnName, String), (ColumnName, SodaType)] = {
+    case (col, rawType) if nonDeprecated(rawType) =>
+      col -> typeMap.getOrElse(rawType, throw new UnknownTypeException(rawType))(col.toString)
```

Comment:
In practice, it doesn't work smooth enough in an expression like `map.flatMap(fnOpt)` because `option2Iterable` doesn't kick in. So I convert expressions to for comprehensions